buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild/propagated-env-variables.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild
+
+import org.gradle.api.tasks.testing.Test
+import gradlebuild.basics.BuildEnvironment
+
+
+val propagatedEnvAllowList = listOf(
+    "OS",
+    "CLASSPATH",
+    "ANDROID_HOME",
+    "DIRNAME",
+    "USERNAME",
+    "USERDOMAIN",
+    "REPO_MIRROR_URL",
+    "APP_HOME",
+    "TMPDIR",
+    "TMP",
+    "TEMP"
+)
+
+
+fun Test.configurePropagatedEnvVariables() {
+    if (BuildEnvironment.isCiServer) {
+        environment(System.getenv().entries.map(::sanitize).toMap())
+    }
+}
+
+
+private
+fun sanitize(entry: MutableMap.MutableEntry<String, String>): Pair<String, String> {
+    return when {
+        entry.key in propagatedEnvAllowList -> entry.key to entry.value
```

Comment:
Can you add some reasons why we pass some of these env-vars? For example I don't understand why we need
- `CI`: The tests should not react to this flag
- `GRADLE_`: What should the tests do with these flags?
- `GIT_`: What are these for?
- `PROCESSOR_`: What is this?
- `TEAMCITY_`, `BUILD_`: The tests shouldn't use this, right?